### ISSUE TITLE
sf project deploy quick command is not giving the deployment job id of quick deployment

### ISSUE THREAD

**Issue by shabu74** [forcedotcom/cli#2415]
### Summary
sf project deploy quick command issue returns successful json response, but with old job id of validation which is same as is the input.

### Steps To Reproduce

Execute sf project deploy start command with --async and get the job id 
Execute sf project deploy quick command (async) passing above job id

### Expected result

job id returned by sf project deploy quick command must be new job id of quick deployment job

### Actual result
ob id returned by sf project deploy quick command is same old job id returned by previous sf project deploy start command

### System Information
{
  "cliVersion": "@salesforce/cli/2.4.10",
  "architecture": "linux-x64",
  "nodeVersion": "node-v19.9.0",
  "osVersion": "Linux 5.15.0-1040-aws",
  "shell": "bash",
  "rootPath": "/usr/local/lib/node_modules/@salesforce/cli",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 2.3.6 (core)",
    "@oclif/plugin-commands 2.2.21 (core)",
    "@oclif/plugin-help 5.2.16 (core)",
    "@oclif/plugin-not-found 2.3.36 (core)",
    "@oclif/plugin-plugins 3.2.0 (core)",
    "@oclif/plugin-search 0.0.22 (core)",
    "@oclif/plugin-update 3.1.30 (core)",
    "@oclif/plugin-version 1.3.8 (core)",
    "@oclif/plugin-warn-if-update-available 2.0.46 (core)",
    "@oclif/plugin-which 2.2.29 (core)",
    "@salesforce/cli 2.4.10 (core)",
    "apex 2.3.10 (core)",
    "auth 2.8.12 (core)",
    "data 2.5.5 (core)",
    "deploy-retrieve 1.17.2 (core)",
    "info 2.6.38 (core)",
    "limits 2.3.29 (core)",
    "login 1.2.25 (core)",
    "org 2.9.30 (core)",
    "schema 2.3.22 (core)",
    "settings 1.4.24 (core)",
    "sobject 0.2.2 (core)",
    "source 2.10.30 (core)",
    "telemetry 2.2.5 (core)",
    "templates [55](https://gitlab.service.nsw.gov.au/salesforce/caresnsw/-/jobs/7718564#L55).5.9 (core)",
    "trust 2.5.4 (core)",
    "user 2.3.28 (core)"
  ]
}
### Additional information
Response of quick command:
{
  "status": 0,
  "result": {
    "id": "0Af9h000006eH9xCAE",
    "done": false,
    "status": "Queued",
    "files": []
  },
  "warnings": [
    "org-api-version configuration overridden at 58.0"
  ]
}

0Af9h000006eH9xCAE is the deployment id of valiodation job, which I passed as input for quick command

**Comment by DaveHahn-Fuel** [comment]
+1 This affects me as well, and is forcing me stick with the sfdx commands

**Comment by mcarvin8** [comment]
Confirmed this affects me as well. My CI job ends early when the validation completes successfully and the quick-deploy starts since it's polling the Validate Job ID for the result.

`const result = await poll(org, jobId, flags.wait);`

Not an observable issue in Sandbox Deployments since Quick-Deploys take about a minute to run. This is easier to see in Production Deployments since Quick-Deployments will still take about a half-hour to run.

**Comment by mdonnalley** [comment]
A [fix](https://github.com/salesforcecli/plugin-deploy-retrieve/pull/748) for this is available in 2.10.2 (currently latest-rc) if you want to give that a try.

**Comment by mcarvin8** [comment]
Thanks. Just upgraded to 2.10.2 and confirmed it fixes this issue.

**Comment by shabu74** [comment]
Yes. Now quick deployment command return the id of new deployment job. But there is another issue:

According to SF documentation of sf project deploy resume:

 "Use this command to resume watching a deploy operation if the original command times out or you specified the --async flag. Deploy operations include standard deploys, quick deploys, deploy validations, and deploy cancellations." 

I have used the deployment id of the quick deploy in resume command. But I am getting error even after successful deployment: 

Error (1): No job found for ID: 0Af9j000006QYQnCAO.

I have checked SF org. 0Af9j000006QYQnCAO is the new deployment job triggered by by quick deploy command and that deployment has been successfully completed.

So, when using resume what I am expecting is either the progress of quick deployment or a completed status. But it is saying no job.

@mdonnalley
